projects/libs/core-library-typescript/src/common-library/grpc-server.ts

```diff
@@ -51,19 +69,54 @@ export class GRPCServerBuilder {
     return this;
   }
 
+  withStartupCheckRunner(startupCheckRunner: StartupCheckRunner) {
+    this.startupCheckRunner = startupCheckRunner;
+    return this;
+  }
+
+
   start(): Promise<any> {
-    const all = Promise.all(this.services.map(s => s.start()));
 
-    this.mali.start(this.endpoint);
 
-    return all;
+    if (!this.endpoint) {
+      return Promise.reject("No endpoint defined, you must call onEndpoint() before starting");
+    }
+
+    if (!this.managementPort) {
+      return Promise.reject("No management port defined, you must call withManagementPort() before starting");
+    }
+
+    if (!this.mali) {
+      return Promise.reject("Mali was not set up correctly. did you forget to call withService()?");
+    }
+
+    // TODO extract to method
+
+    const app = express();
+    app.get("/admin/startupCheck", (req: Request, res: Response) => {
+      this.startupCheckRunner.run()
+        .then((result: StartupStatus) => {
+          const httpCode = result.status === STARTUP_STATUS.OK ? 200 : 503;
+          return res.status(httpCode).send(result);
+        });
+    });
+
+    return new Promise(resolve => {
+      this.managementServer = app.listen(this.managementPort, resolve);
+    })
+      .then(() => {
+        const all = Promise.all(this.services.map(s => s.start()));
+        this.mali.start(this.endpoint);
+        return all;
+
+
+      });
   }
 
   stop(): Promise<any> {
     const all = Promise.all(this.services.map(s => s.stop()));
-
-    this.mali.close(this.endpoint);
-
+    this.mali && this.mali.close(this.endpoint);
```

Comment:
Please use if statement here.

Reply:
Ok, although this is identical to "if", and is just as readable. 
I guess for V1 we may want to discuss code style more thoroughly. I'm fine with either method.